src/main/java/org/cyclonedx/util/LicenseResolver.java

```diff
@@ -161,32 +186,26 @@ private static LicenseChoice createLicenseChoice(String licenseId, String primar
         return choice;
     }
 
-    /**
-     * Attempts to perform high-confidence license resolution with unstructured text as input.
-     * @param licenseString the license string (not the actual license text)
-     * @param includeLicenseText specifies is the resolved license will include the entire text of the license
-     * @return a LicenseChoice object if resolved, otherwise null
-     */
-    private static LicenseChoice resolveViaAlternativeMapping(String licenseString, boolean includeLicenseText) throws IOException {
-        if (licenseString == null) {
-            return null;
-        }
-        for (final Map.Entry<String, List<String>> mapping : mappings.entrySet()) {
-            final List<String> names = mapping.getValue();
-            if (names != null) {
-                for (final String name: names) {
-                    if (licenseString.equalsIgnoreCase(name)) {
-                        if (mapping.getKey().startsWith("(") && mapping.getKey().endsWith(")")) {
-                            final LicenseChoice lc = new LicenseChoice();
-                            lc.setExpression(mapping.getKey());
-                            return lc;
-                        } else {
-                            return createLicenseChoice(mapping.getKey(), null, false, includeLicenseText);
-                        }
-                    }
-                }
-            }
-        }
-        return null;
+    private static class LicenseDetail {
```

Comment:
Helper classes so that you a) use Jackson to parse the json that's coming in, b) make all your loops a lot easier, and type checking